test/anyone-can-renew-test.js

```diff
@@ -0,0 +1,331 @@
+    );
+    assert.strictEqual(
+      address.toString('regtest'),
+      'rs1qu3nrzrjkd783ftpk7l4hvpa96aazx5dddw66hgs2zuukckcchrqs570axm'
+    );
+    assert.strictEqual(
+      address.toString('simnet'),
+      'ss1qu3nrzrjkd783ftpk7l4hvpa96aazx5dddw66hgs2zuukckcchrqs4kzusf'
+    );
+  });
+
+  it('should fund both wallets', async () => {
+    await mineBlocks(2, aliceReceive);
+    await mineBlocks(2, bobReceive);
+
+    await forValue(wdb, 'height', node.chain.height);
+
+    const aliceBal = await alice.getBalance();
+    const bobBal = await bob.getBalance();
+    assert(aliceBal.confirmed === 2000 * 2 * 1e6);
+    assert(bobBal.confirmed === 2000 * 2 * 1e6);
+  });
+
+  it('should win name with Alice\'s wallet', async () => {
+    heightBeforeOpen = node.chain.height;
+
+    await alice.sendOpen(name, false);
+    await mineBlocks(network.names.treeInterval + 1);
+
+    await alice.sendBid(name, 100000, 200000);
+    await mineBlocks(network.names.biddingPeriod);
+
+    await alice.sendReveal(name);
+    await mineBlocks(network.names.revealPeriod + 1);
+
+    let ns = await node.getNameStatus(nameHash);
+    assert(ns);
+    const owner = ns.owner;
+    const coin = await alice.getCoin(owner.hash, owner.index);
+    assert(coin);
+    const json = ns.getJSON(node.chain.height, node.network);
+    assert(json.state === 'CLOSED');
+
+    heightBeforeRegister = node.chain.height;
+
+    const resource = Resource.fromJSON({
+      records: [{type: 'TXT', txt: ['This name is ANYONE-CAN-RENEW']}]
+    });
+    await alice.sendUpdate(name, resource);
+    await mineBlocks(network.names.treeInterval);
+
+    ns = await node.getNameStatus(nameHash);
+    assert.strictEqual(ns.height, heightBeforeOpen + 1);
+    assert.strictEqual(ns.renewal, heightBeforeRegister + 1);
+  });
+
+  it('should TRANSFER/FINALIZE to ANYONE-CAN-RENEW address', async () => {
+    const heightBeforeTransfer = node.chain.height;
+
+    await alice.sendTransfer(name, address);
+    await mineBlocks(network.names.transferLockup);
+
+    let ns = await node.getNameStatus(nameHash);
+    assert.strictEqual(ns.transfer, heightBeforeTransfer + 1);
+
+    heightBeforeFinalize = node.chain.height;
+
+    await alice.sendFinalize(name);
+    await mineBlocks(1);
+
+    // FINALIZE resets transfer and renewal
+    ns = await node.getNameStatus(nameHash);
+    assert.strictEqual(ns.transfer, 0);
+    assert.strictEqual(ns.height, heightBeforeOpen + 1);
+    assert.strictEqual(ns.renewal, heightBeforeFinalize + 1);
+
+    const {hash, index} = ns.owner;
+    coin = await node.getCoin(hash, index);
+    assert.deepStrictEqual(coin.address, address);
+  });
+
+  it('should not be owned by either wallet', async  () => {
+    assert.rejects(
+      alice.sendTransfer(name, aliceReceive),
+      {message: `Wallet does not own: "${name}".`}
+    );
+
+    assert.rejects(
+      bob.sendTransfer(name, bobReceive),
+      {message: 'Auction not found.'}
```

Comment:
Just curious why this says 'Auction not found' instead of `Wallet does not own` above.

Reply:
It's because Alice sent the open, the bid, the transfer - everything. Bob is just an innocent bystander (aka "anyone") so his wallet has no namestate data at all. Alice's wallet has some namestate data because she bid and actually owned the name at some point in the past. So its like, in her "watchlist" (walletDB `nameMap`). But the UTXO that owns the name is not in her wallet or under her control.

You did make me curious though Im going to add a test to see if this action causes bob's wallet to start tracking the name, especially now that #564 has been merged.

Reply:
ok updated, Alice's wallet still tracks the name but Bob's never does.